### DISCUSSION TITLE
Alphatensor matrix multiplication algorithms

### DISCUSSION THREAD

**Discussion by jamen** [bitshifter/glam-rs#359] [Ideas]
You've probably seen [Alphatensor discovering new matrix multiplication algorithms](https://www.deepmind.com/blog/discovering-novel-algorithms-with-alphatensor). Is there interest in porting these to glam?

**Comment by bitshifter** [comment]
I hadn't seen that, very interesting. I would potentially be interested in seeing the scalar implementations. They may or may not translate well to SIMD. I only skimmed the paper but I wonder if they took floating point accuracy into account when coming up with new algorithms.

**Comment by jamen** [comment]
Forgot to link https://github.com/drbh/simd-alphatensor-rs. Here is [mat4x4 * mat4x4](https://github.com/drbh/simd-alphatensor-rs/blob/main/src/gen.rs#L5596)

**Comment by bitshifter** [comment]
Hmm, they're only benchmarking with `i32`, it would be interesting to see `f32`.
